mantidimaging/gui/windows/live_viewer/model.py

```diff
@@ -137,6 +214,46 @@ def close(self) -> None:
             self.image_watcher = None
         self.presenter = None  # type: ignore # Model instance to be destroyed -type can be inconsistent
 
+    def add_mean(self, image_data_obj: Image_Data, image_array: np.ndarray) -> None:
+        if self.roi and (self.roi.left, self.roi.top, self.roi.right, self.roi.bottom) != (0, 0, 0, 0):
+            left, top, right, bottom = self.roi
+            mean_to_add = np.mean(image_array[top:bottom, left:right])
+        else:
+            mean_to_add = np.mean(image_array)
+        self.mean_dict[image_data_obj.image_path] = mean_to_add
+        self.mean = np.append(self.mean, mean_to_add)
+
+    def clear_mean_partial(self) -> None:
+        self.mean_dict.clear()
+        self.mean = np.full(len(self.images), np.nan)
+
+    def clear_mean(self) -> None:
+        self.mean_dict.clear()
+        self.mean = np.delete(self.mean, np.arange(self.mean.size))
+
+    def calc_mean_fully(self) -> None:
+        if self.images is not None:
+            for image in self.images:
+                self.add_mean(image, self.image_cache.load_image(image))
+
+    def calc_mean_chunk(self, chunk_size: int) -> None:
+        if self.images is not None:
+            nanInds = np.argwhere(np.isnan(self.mean))
+            if self.roi:
+                left, top, right, bottom = self.roi
+            else:
+                left, top, right, bottom = (0, 0, -1, -1)
+            if nanInds.size > 0:
+                for ind in range(len(nanInds) - 1, len(nanInds) - 1 - chunk_size, -1):
+                    if ind < 0:
+                        break
```

Comment:
Would `for ind in nanInds[-1:-chunk_size:-1]:` work?
Or `for ind in nanInds.flip()[:chunk_size]:`

Reply:
`for ind in nanInds[-1:-chunk_size:-1]` works!